atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/cleaner/KeyValueServicePuncherStoreTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.cleaner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.keyvalue.impl.InMemoryKeyValueService;
+
+public class KeyValueServicePuncherStoreTest {
+    private static final long TIMESTAMP_1 = 10L;
+    private static final long TIMESTAMP_2 = 20L;
+    private static final long TIMESTAMP_3 = 30L;
+    private static final long TIMESTAMP_BETWEEN_1_AND_2 = 15L;
+
+    private static final long WALL_CLOCK_1 = 100L;
+    private static final long WALL_CLOCK_2 = 200L;
+    private static final long WALL_CLOCK_3 = 300L;
+    private static final long WALL_CLOCK_BETWEEN_1_AND_2 = 150L;
```

Comment:
smartassed nit: could use `(WALL_CLOCK_1 + WALL_CLOCK_2) / 2` (and analogously for the other two "between" constants.